### DISCUSSION TITLE
Scrolling over grid

### DISCUSSION THREAD

**Discussion by BctfN0HUK7Yg** [emilk/egui#908] [Q&A] [answered]
Hi.
I'm tying to control scrollingArea and have a problem.
There is a grid with a lot of cells. It contains in ScrollArea. Scrollbars work right. 
I want to scroll by holding middle mouse button and move over the grid. How to do this?

How to get response for ScrollArea to check such draging?
How to move content of ScrollArea? ( I found pub fn scroll_offset(mut self, offset: f32) -> Self { self.offset = Some(Vec2::new(0.0, ffset));  self } But it works only horisontally.

**Comment by BctfN0HUK7Yg** [comment, marked as answer]
Ok, I found this

               let mut scroll_area = egui::containers::ScrollArea::both();
               if ui.input().pointer.middle_down() { self.y_offset-=ui.input().pointer.delta().y}
                 scroll_area = scroll_area.scroll_offset(self.y_offset); 
               scroll_area.show(ui, |ui| { ....

but how to scroll horizontally? In the demo there is vertical_scroll method, but i hasn't it

**Reply by emilk** [reply]
Perhaps you need to update `egui` - horizontal scrolling was added in 0.15.0: https://github.com/emilk/egui/blob/master/CHANGELOG.md#0150---2021-10-24---syntax-highlighting-and-hscroll
